### ISSUE TITLE
How to pass attention_mask?

### ISSUE THREAD

**Issue by ghost** [LIYUESEN/druggpt#10]
'The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results'

I am getting this warning after every batch. How can I specify this attention_mask, or can I safely ignore this?

Running on Ubuntu 22.04 LTS.

Many thanks/Evert

**Comment by LIYUESEN** [comment]
Dear Evert,

Thanks for reaching out. I've noted the problem you mentioned. The attention_mask is an optional input in the transformers library, serving to mask out filled values in the sequence. When generating outputs, if all of your input sequences have the same length and you have not filled any sequences, then you can safely ignore this warning. In our case, we are consistently dealing with non-padded, fixed-length sequences for subsequent generation, so this warning can be safely ignored. The warning can be eliminated by creating 'attention_mask' and specifying 'pad_token_id', and we can improve this issue after further testing.

Best,
Yuesen Li

![image](https://github.com/LIYUESEN/druggpt/assets/40382239/f4fb9175-a554-481c-910d-0687812d8df5)

**Comment by ghost** [comment]
Dear Yuesen,

Thanks for your reply, I will ignore it for now.

On a different note:

I started a job on a small druggable protein using the -n 50 option. How many batches will be generated? The job has been running for 4 h and reached almost 500 batches.

BW/Evert

**Comment by LIYUESEN** [comment]
Hi Evert,

Thanks for raising this concern. I've also noticed this issue and made some corresponding updates. However, since this seems to be a different problem, could you please open a new issue for this? I would be glad to help and respond to your question there.

Best,
Yuesen Li
